### PR TITLE
EVG-8299 wrap ssh command for hosts

### PR DESCRIPTION
**Pull request by ablack12**
My only open question is if the key flag should also accept "-i" (because that's what the actual ssh flag is called so maybe what people would expect) or if we want to stick to Evergreen conventions, what do you think @ybrill?

### CODE REVIEW

File: operations/host_spawn.go [hunk 1]
```diff
@@ -480,6 +480,58 @@ func hostStart() cli.Command {
 	}
 }
 
+func hostSSH() cli.Command {
+	const (
+		keyFlagName = "key"
```
Comment (ybrill) [inline operations/host_spawn.go:485]: I don't understand what you mean by Evergreen conventions. Do we usually use `-i ` for something else?

Comment (ablack12) [reply operations/host_spawn.go:485]: Ah what I mean is that in many host routes, we use "key" or "k" to reference the ssh key, with the message "name or value of an public key to use".  For ssh, "-i" is "Selects a file from which the identity (private key) for public key authentication is read." We could instead say something like "path to the private key (reference -i for ssh)

File: operations/host_spawn.go [hunk 2]
```diff
@@ -480,6 +480,58 @@ func hostStart() cli.Command {
...
+				return errors.Wrap(err, "problem getting host")
+			}
+			if h == nil {
+				return errors.New("host not found")
+			}
+			if h.Status == nil || *h.Status != evergreen.HostRunning {
+				return errors.New("host is not running")
+			}
+			if h.User == nil || h.HostURL == nil {
+				return errors.New("unable to ssh into host without user or DNS name")
```
[the remark this answers is not in the document]
Comment (ablack12) [reply operations/host_spawn.go:524]: Sure I forgot about that!

File: operations/host_spawn.go [hunk 3]
```diff
@@ -480,6 +480,58 @@ func hostStart() cli.Command {
...
+			}
+			args := []string{"ssh", "-tt", fmt.Sprintf("%s@%s", *h.User, *h.HostURL)}
```
[the remark this answers is not in the document]
Comment (ablack12) [reply operations/host_spawn.go:526]: Good question--without it I was getting an error, and found the solution [here](https://stackoverflow.com/questions/7114990/pseudo-terminal-will-not-be-allocated-because-stdin-is-not-a-terminal). To be honest, I only somewhat understand the error; I think that this is considered less safe so they want to make sure it's what you expect?

[the remark this answers is not in the document]
Comment (ablack12) [reply operations/host_spawn.go:526]: Ah I didn't realize that. Do you know how to verify if ssh is on the path? If it's not on the path then it will return an error to the user right? I would expect users to know that ssh should be on the path by this point.

File: operations/host_spawn.go (hunk 1 above)
Comment (ablack12) [reply operations/host_spawn.go:485]: I just realized typing that out that -k is for the public key, so it's probably okay to change this to -i.

File: operations/admin.go [hunk 4]
```diff
@@ -497,7 +497,7 @@ func adminBackup() cli.Command {
 			serviceConfigFlags(),
 			addDbSettingsFlags(
 				cli.StringFlag{
-					Name:  joinFlagNames(collectionNameFlag, "c"),
+					Name:  collectionNameFlag,
```
Comment (ablack12) [inline operations/admin.go:500]: This is unrelated: in looking for examples of Value for flags (which I've decided not to use) I tried `evergreen admin backup -h` which panics because -c is repeated, so I figured I would just fix it.

File: operations/host_spawn.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (bsamek) [reply operations/host_spawn.go:526]: I think we can assume people have ssh on their PATH. Certainly the macbooks we provide and any standard Linux distro someone might be using would have it. Also, if they don't have it, I think the error would be clear.

File: operations/host_spawn.go [hunk 5]
```diff
@@ -480,6 +480,58 @@ func hostStart() cli.Command {
...
+			defer cancel()
+
+			conf, err := NewClientSettings(confPath)
+			if err != nil {
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			client := conf.setupRestCommunicator(ctx)
+			defer client.Close()
+
+			h, err := client.GetSpawnHost(ctx, hostID)
```
Comment (bsamek) [inline operations/host_spawn.go:513]: Since we allow people to name hosts, should we also look for the host by name? Also, if someone only has one workstation, could we return that one?

Comment (ybrill) [reply operations/host_spawn.go:513]: we don't (yet) validate the hosts' names are unique

Comment (ablack12) [reply operations/host_spawn.go:513]: I also don't want to just return the one workstation, since this code isn't technically workstation specific. But ids don't change so users could easily make an alias for their workstation.

File: operations/host_spawn.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (ybrill) [reply operations/host_spawn.go:526]: neat.
An alternative solution that occurred to me was to `bash -c "ssh user@host"`.

File: operations/host_spawn.go [hunk 6]
```diff
@@ -480,6 +480,58 @@ func hostStart() cli.Command {
 	}
 }
 
+func hostSSH() cli.Command {
+	const (
+		identityFlagName = "identity"
```
Comment (ybrill) [inline operations/host_spawn.go:485]: should this be called `identity_file`?

File: operations/host_spawn.go [hunk 7]
```diff
@@ -480,6 +480,58 @@ func hostStart() cli.Command {
...
+			if h == nil {
+				return errors.New("host not found")
+			}
+			if restModel.FromStringPtr(h.Status) != evergreen.HostRunning {
+				return errors.New("host is not running")
+			}
+			if restModel.FromStringPtr(h.User) == "" || restModel.FromStringPtr(h.HostURL) == "" {
+				return errors.New("unable to ssh into host without user or DNS name")
+			}
+			args := []string{"ssh", "-tt", fmt.Sprintf("%s@%s", *h.User, *h.HostURL)}
```
Comment (ybrill) [inline operations/host_spawn.go:526]: should these (`h.User` and `h.HostURL`) use FromStringPtr as well?

File: operations/host_spawn.go (hunk 6 above)
Comment (ablack12) [reply operations/host_spawn.go:485]: I think that's long to type out.

Comment (ablack12) [reply operations/host_spawn.go:485]: Ah but I see that is the ssh name :( So i guess yes haha

File: operations/host_spawn.go (hunk 7 above)
Comment (ablack12) [reply operations/host_spawn.go:526]: I can change it, although I'd argue it's not necessary since  we already know they aren't nil.

Comment (ybrill) [reply operations/host_spawn.go]: agreed

File: operations/host_spawn.go [hunk 8]
```diff
@@ -480,6 +480,60 @@ func hostStart() cli.Command {
...
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			client := conf.setupRestCommunicator(ctx)
+			defer client.Close()
+
+			h, err := client.GetSpawnHost(ctx, hostID)
+			if err != nil {
+				return errors.Wrap(err, "problem getting host")
+			}
+			if h == nil {
```
Comment (ybrill) [inline operations/host_spawn.go:517]: I think `GetSpawnHost` will return an error if the host isn't found
